AdyenActions/Components/BACS/BACSActionComponent.swift

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import AdyenNetworking
+
+/// A component that handles actions after paying with BACS direct debit.
+public final class BACSActionComponent: ActionComponent, ShareableComponent {
+    /// :nodoc:
+    public let apiContext: APIContext
+    
+    /// :nodoc:
+    public weak var delegate: ActionComponentDelegate?
+    
+    /// Delegates `PresentableComponent`'s presentation.
+    public weak var presentationDelegate: PresentationDelegate?
+    
+    /// The Component UI style.
+    public let style: BACSActionComponentStyle
+    
+    /// :nodoc:
+    public var localizationParameters: LocalizationParameters?
+    
+    /// :nodoc:
+    internal let presenterViewController = UIViewController()
```

Comment:
I think you missed the same property is in VoucherComponent :D
I created this protocol to prevent duplication and combined the sharing functionality that both `VoucherComponent` and `BACSActionComponent` has into a protocol.
```
internal protocol ShareableComponent: AnyObject {
    var presenterViewController: UIViewController { get }
    
    func setUpPresenterViewController(parentViewController: UIViewController)
    
    func presentSharePopover(with item: Any, sourceView: UIView)
}
```

Reply:
Maybe, can't check the code at the moment, but if it's in VoucherComponent, then it limits the reusability.

If something is in old code, doesn't mean it's the best way to go, right?

Reply:
Of course not, I only meant so since you gave the example of Voucher.
let's discuss that property when you come back 👍